client/my-sites/invites/invite-accept/index.jsx

```diff
@@ -49,32 +49,34 @@ class InviteAccept extends React.Component {
 		matchEmailError: false,
 	};
 
-	UNSAFE_componentWillMount() {
+	componentDidMount() {
 		// The site ID and invite key are required, so only fetch if set
 		if ( this.props.siteId && this.props.inviteKey ) {
-			fetchInvite( this.props.siteId, this.props.inviteKey );
+			this.fetchInvite();
 		}
-
-		InvitesStore.on( 'change', this.refreshInvite );
-	}
-
-	componentWillUnmount() {
-		InvitesStore.off( 'change', this.refreshInvite );
 	}
 
-	refreshInvite = () => {
-		const invite = InvitesStore.getInvite( this.props.siteId, this.props.inviteKey );
-		const error = InvitesStore.getInviteError( this.props.siteId, this.props.inviteKey );
+	async fetchInvite() {
+		try {
+			const response = await wpcom
+				.undocumented()
+				.getInvite( this.props.siteId, this.props.inviteKey );
 
-		if ( invite ) {
-			// add subscription-related keys to the invite
-			Object.assign( invite, {
+			const invite = {
+				...normalizeInvite( response ),
 				activationKey: this.props.activationKey,
 				authKey: this.props.authKey,
+			};
+
+			this.setState( { invite } );
+		} catch ( error ) {
+			this.setState( { error } );
+
+			recordTracksEvent( 'calypso_invite_validation_failure', {
```

Comment:
Would be nice if we could port this to use the Redux `recordTracksEvent` instead, too. Not a requirement for this PR though; can be done in another one.